service-base/src/main/java/org/eclipse/hono/service/tenant/BaseTenantService.java

```diff
@@ -162,15 +162,27 @@ public final void processTenantMessage(final Message<JsonObject> tenantMsg) {
                 remove(tenantId, result -> reply(tenantMsg, result));
                 break;
             default:
-                log.debug("invalid action in request message [{}]", subject);
-                reply(tenantMsg, TenantResult.from(HTTP_BAD_REQUEST));
+                processCustomTenantMessage(tenantMsg, subject);
+                break;
             }
         } catch (final ClassCastException e) {
             log.debug("malformed request message [{}]", e.getMessage());
             reply(tenantMsg, TenantResult.from(HTTP_BAD_REQUEST));
         }
     }
 
+    /**
+     * Override the following method to extend TenantService with implementation
+     * specific operations.
+     *
+     * @param tenantMsg target tenant message to be processed.
+     * @param action implementation specific action to be executed.
+     */
+    protected void processCustomTenantMessage(final Message<JsonObject> tenantMsg, String action) {
+        log.debug("invalid action in request message [{}]", action);
```

Comment:
there are three variants of this log statement. Can we please use the same one for all three services?

Reply:
will do it.

Reply:
Done